apps/05Mastermind.js

```diff
@@ -26,14 +28,52 @@ function getRandomInt(min, max) {
   return Math.floor(Math.random() * (max - min)) + min;
 }
 
-function generateHint() {
+function generateHint(solution, guess) {
     // your code here
+    var solutionArray = solution.split("");
+    var guessArray = guess.split("");
+    var correctLetterLocations = 0;
+    var correctLetters = 0;
+
+    for (index= 0; index < guessArray.length; index++) {
+      if (guessArray[index] === solutionArray[index]) {
+        correctLetterLocations++;
+        solutionArray[index] = null;
+      }
+    }
+
+    for (index= 0; index < guessArray.length; index++) {
+    var targetIndex = solutionArray.indexOf(guessArray[index]);
+    if (guessArray.indexOf(solutionArray[index]) > -1) {
+      correctLetters++;
+      solutionArray[index] = null;
```

Comment:
Hmmm something about this line looks a little off to me. Shouldn't we be using the `targetIndex` variable?